airflow/providers/amazon/aws/sensors/eks.py

```diff
@@ -57,7 +58,71 @@
 )
 
 
-class EksClusterStateSensor(BaseSensorOperator):
+class EksBaseSensor(BaseSensorOperator):
+    """
+    Base class to check various EKS states.
+    Subclasses need to implement get_state and get_terminal_states methods.
+
+    :param cluster_name: The name of the Cluster
+    :param target_state: Will return successfully when that state is reached.
+    :param target_state_type: The enum containing the states,
+        will be used to convert the target state if it has to be converted from a string
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+        If this is None or empty then the default boto3 behaviour is used. If
+        running Airflow in a distributed manner and aws_conn_id is None or
+        empty, then the default boto3 configuration would be used (and must be
+        maintained on each worker node).
+    :param region: Which AWS region the connection should use.
+        If this is None or empty then the default boto3 behaviour is used.
+    """
+
+    def __init__(
+        self,
+        *,
+        cluster_name: str,
+        target_state: Any,
```

Comment:
I think I would prefer to see this as 
```suggestion
        target_state: ClusterStates | NodegroupStates | FargateProfileStates,
```

and modify the return type hinting on the get_state() methods below, but I can also see how that would be getting a bit too verbose, so I'll leave it up to you.  Maybe a new Type:

```
from typing import NewType
TargetStates = NewType('TargetStates', ClusterStates | NodegroupStates | FargateProfileStates)
```

here or in the hook?  Just a thought (non-blocking)

Reply:
I would have liked to write a class with generics here, but python doesn't allow that :(

Reply:
In python you can do it like this:

```
from typing import TypeVar

T = TypeVar('T')      # Declare type variable

def first(l: Sequence[T]) -> T:   # Generic function
    return l[0]
```

But at that point I honestly don't know how that is better than just using 'Any'.

Reply:
the problem is that I think you can templatize a function but not a class.